specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2020-12-01/Marketplace.json

```diff
@@ -0,0 +1,1068 @@
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PlanNotificationDetails"
+          }
+        }
+      }
+    },
+    "StopSellNotifications": {
+      "description": "Stop sell notification details",
+      "type": "object",
+      "properties": {
+        "offerId": {
+          "description": "Gets offer id",
+          "type": "string"
+        },
+        "displayName": {
+          "description": "Gets offer display name",
+          "type": "string"
+        },
+        "isEntire": {
+          "description": "Gets a value indicating whether entire offer is in stop sell or only few of its plans",
+          "type": "boolean"
+        },
+        "messageCode": {
+          "description": "Gets or sets the notification message id",
+          "type": "integer",
+          "format": "int64"
+        },
+        "icon": {
+          "description": "Gets or sets the icon url",
+          "type": "string"
+        },
+        "plans": {
+          "description": "Gets or sets removed plans notifications",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PlanNotificationDetails"
+          }
+        }
+      }
+    },
+    "PlanNotificationDetails": {
+      "description": "Plan notification details",
+      "type": "object",
+      "properties": {
+        "planId": {
+          "description": "Gets or sets the plan id",
+          "type": "string"
+        },
+        "planDisplayName": {
+          "description": "Gets or sets the plan display name",
+          "type": "string"
+        }
+      }
+    },
+    "RequestApprovalsDetails": {
+      "description": "Request approvals details",
+      "type": "object",
+      "properties": {
+        "offerId": {
+          "description": "Gets offer id",
+          "type": "string"
+        },
+        "displayName": {
+          "description": "Gets offer display name",
+          "type": "string"
+        },
+        "publisherId": {
+          "description": "Gets or sets publisher id",
+          "type": "string"
+        },
+        "messageCode": {
+          "description": "Gets or sets the notification message id",
+          "type": "integer",
+          "format": "int64"
+        },
+        "icon": {
+          "description": "Gets or sets the icon url",
+          "type": "string"
+        },
+        "plans": {
+          "description": "Gets or sets removed plans notifications",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PlanNotificationDetails"
+          }
+        }
+      }
+    },
+    "AdminRequestApprovalsList": {
+      "description": "List of admin request approval resources",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AdminRequestApprovalsResource"
+          }
+        },
+        "nextLink": {
```

Comment:
Make sure to mark nextLink as readOnly as well. Review your properties and mark them as readOnly where is needed.

Reply:
done